src/bot/commands/shoutout.js

```diff
@@ -1,21 +1,36 @@
 import { command } from '../lib/utils/command.js'
 
-function resolver({ command: { args, memory }, isModerator, isBroadcaster }) {
-  const [username] = args
+function resolver(context) {
+  const {
+    command: {
+      args: [username],
+    },
+    isModerator,
+    isBroadcaster,
+  } = context
+  let {
+    memory: { shoutout },
+  } = context
 
   if (username != null) {
     if (isModerator !== true && isBroadcaster !== true) {
       return `Please STOP. You're not a moderator! mindse4Stop`
-    } else {
-      memory.shoutout = username
     }
+
+    shoutout = username.replace(`@`, ``)
   }
 
-  if (memory.shoutout == null) {
+  if (shoutout == null) {
     return ``
   }
 
-  return `Follow @${memory.shoutout} over at twitch.tv/${memory.shoutout} ! <3 Kreygasm`
+  return {
+    ...context,
+    memory: {
+      shoutout,
```

Comment:
Instead of memorized the last !so username and reused it for all new !so, should we improve the memory to use "by user" !so.

If I do !so mindsers, it will memorize that my so is for mindsers and repeated !so shoutout the same. But if someone else do !so maeva, this someone else next so will be redirected to maeva and mine will stay at mindsers.

This will add some complexity but will be cool to multi-ads

Reply:
So every staff member (moderator + streamer) will have his own shoutout memory?

Why not having a list of account to shoutout. Every time someone call `!so` it randomly shout out one account of the list.

Reply:
If we do this kind of implementation, we must provide a way to "remove" someone from this list. Imagine if a user is !so an is ban from the chat. The !so command can possibly use this user again.

Reply:
Fair point. Need to think about it